/*  try demo 1*/#build CompilerOptions @"-Wno-unused-but-set-variable"// silence compiler warning for unused variable 'obj'include "NSLog.incl"void local fn TestCFStringRef  obj = NULLExceptionRef eCFArrayRef array = @[@"Alpha",@"Bravo",@"Charlie",@"Echo"]tryNSLog(@"-try-\ntry to get object at index 4 of array")// the array has only been given four elements (0-3) so trying to access any element above 3 will throw an exception// and probably crash the app had this call not been inside the try blockobj = array[4] // if exception, skip the rest of this block and go straight to catch --->// we only get here if there was no exceptionNSLog(@"no exception")end try// !!! don't put any code betwen try and catch blockscatch ( e ) // <--- we arrive here if exception in try block// the value assigned to 'e', is only accessible from inside the catch blockNSLog(@"\n-catch-\n%@: %@",fn ExceptionName(e),fn ExceptionReason(e))end catch// !!! don't put any code betwen catch and finally blocksfinally// code inside the finally block is always executed whether an exception is thrown or notNSLog(@"\n-finally-\ncleanup")// ... cleanupend finally// ... continueend fnfn TestHandleEvents